pennylane/pauli/pauli_arithmetic.py

```diff
@@ -257,19 +267,34 @@ def to_mat(self, wire_order, format="dense"):
         Rasies:
             ValueError: Can't get the matrix of an empty PauliSentence.
         """
+
+        def _pw_wires(w: Iterable) -> wires.Wires:
+            """Return the native Wires instance for a list of wire labels.
+            w represents the wires of the PauliWord being processed. In case
+            the PauliWord is empty ({}), choose any arbitrary wire from the
+            PauliSentence it is composed in.
+            """
+            if w:
+                return wires.Wires(w)
+
+            ps_wires = self.wires
+            if len(ps_wires) > 0:
+                return wires.Wires(
+                    list(ps_wires)[0]
+                )  # return any wire from the Pauli sentence's wires
+            return wires.Wires([])
+
         if len(self) == 0:
             if wire_order is None or wire_order == wires.Wires([]):
                 raise ValueError("Can't get the matrix of an empty PauliSentence.")
-            return (
-                np.eye(2 ** len(wire_order))
-                if format == "dense"
-                else sparse.eye(2 ** len(wire_order), format=format)
-            )
+            if format == "dense":
+                return np.eye(2 ** len(wire_order))
+            return sparse.eye(2 ** len(wire_order), format=format)
 
         mats_and_wires_gen = (
             (
-                coeff * pw.to_mat(wire_order=wires.Wires(list(pw.wires)), format=format),
-                wires.Wires(list(pw.wires)),
+                coeff * pw.to_mat(wire_order=_pw_wires(list(pw.wires)), format=format),
+                _pw_wires(list(pw.wires)),
```

Comment:
Why do we need to convert to a list first?

Reply:
We don't need to cast into list here, because the logic:
```
if w:
    return wires.Wires(w)
```
can be done with sets.

Reply:
And `Wires._labels` is a tuple/ list anyway...

Reply:
But `pw.wires` is a `set`, not a `Wires` object

Reply:
Updated 👍🏼

Reply:
@AlbertMitjans It's kind of a hybrid. Certain things use a set representation and others use a tuple/ list representation.